platform-sdk/docs/proposals/TSS-Encryption-Key-Automation/TSS-Encryption-Key-Automation.md

```diff
@@ -0,0 +1,233 @@
+      private key, a new public/private `tssEncryptionKey` will need to be generated through a call back to
+      `CryptoStatic.java` to generate the key and persist the new private key to disk. The new public key will need
+      to be distributed to the rest of the network as in `1.1` above.
+   2. If the public key in the state matches the loaded private key, all is good.
+
+At the end of the `TssBaseService` initialization there will be two outcomes:
+
+1. The public key in the state matches the private key and the `TssBaseService` is ready to execute normally.
+2. At the start of `TssBaseService` execution, it needs to update the public key on the network and cannot perform
+   its normal execution until the public key has been gossiped, handled, and updated in the state.
+
+#### TssBaseService Start Of Execution
+
+If there is a public key update needing to happen:
+
+1. Suspend normal execution: The `TssBaseService` will not be able to decrypt its private shares and cannot
+   participate in signing or re-keying.
+2. Send (and periodically resend) a `TssEncryptionKeyTransaction` until the public key is updated in the state.
+3. Normal execution of the `TssBaseService` cannot resume until a new candidate roster has been keyed and adopted
+   that provides private shares to this node that can be decrypted using the new `tssEncryptionKey`.
+
+New `TssEncryptionKeyTransaction` handle workflow:
+
+* Upon receipt of a `TssEncryptionKeyTransaction`:
+  1. Add the public key to the `TssBaseService` state.
+  2. Disable any mechanism created for resending the transaction.
+  3. Check condition for resuming normal execution.
+
+Resuming / Proceeding with Normal Execution:
+
+* Normal execution can resume when private shares can be decrypted from the `TssMessages` belonging to the
+  latest `active roster`.
+
+### Public API
+
+#### TssEncryptionKeyTransaction Protobuf
+
+The `TssEncryptionKeyTransaction` will only appear in events created from the node sending the transaction. The
+signature on the event and the creator identifier of the event are sufficient to provide authenticity of the
+transaction.
+
+```protobuf
+message TssEncryptionKeyTransaction {
+  /**
+   * The raw bytes of the public TSS encryption key of the node sending the transaction.
+   */
+  bytes publicTssEncryptionKey = 1;
+}
+```
+
+#### TssBaseService New State
+
+The `TssBaseService` state will be extended with a new map of the public `tssEncryptionKey` for each node.
+
+1. The keys of the map are Longs/UInt64 for the `nodeId` of the nodes
+2. The values of the map are the latest `TssEncryptionKeyTransaction` from each node.
+
+Data Lifecycle:
+
+* When a node id is not present in any of the `active rosters` or a `candidate roster`, the entry for that node id
+  should be removed. This cleanup task should trigger any time there is a candidate or active roster state change.
```

Comment:
Does the removal of entries need any synchronization and/or transactions between nodes to ensure they end up with identical lists of public keys?